qubes_config/global_config/global_config.py

```diff
@@ -0,0 +1,560 @@
+            policy_manager=self.policy_manager,
+            default_policy="""qubes.OpenInVM * @adminvm @anyvm deny\n
+qubes.OpenInVM * @anyvm @dispvm allow\n
+qubes.OpenInVM * @anyvm @anyvm ask""",
+            service_name="qubes.OpenInVM",
+            policy_file_name="50-config-openinvm",
+            verb_description=TargetedVerbDescription(
+                    single_target_descr={
+                        "allow": 'open files in',
+                        "ask": 'where to open files,\nand select by default',
+                        "deny": 'be allowed to open files in'
+                    },
+                    multi_target_descr={
+                        "allow": 'open files in',
+                        "ask": 'where to open files in',
+                        "deny": 'be allowed to open files in'
+                    }
+                ),
+            rule_class=RuleTargeted)
+
+    def reset(self):
+        self.filecopy_handler.reset()
+        self.openinvm_handler.reset()
+
+    def save(self):
+        self.filecopy_handler.save()
+        self.openinvm_handler.save()
+
+    def get_unsaved(self) -> str:
+        unsaved = []
+        for handler in [self.filecopy_handler, self.openinvm_handler]:
+            unsaved_changes = handler.get_unsaved()
+            if unsaved_changes:
+                unsaved.append(unsaved_changes)
+        return "\n".join(unsaved)
+
+
+class ThisDeviceHandler(PageHandler):
+    """Handler for the ThisDevice page."""
+    def __init__(self,
+                 qapp: qubesadmin.Qubes,
+                 gtk_builder: Gtk.Builder):
+        self.qapp = qapp
+
+        self.model_label: Gtk.Label = gtk_builder.get_object(
+            'thisdevice_model_label')
+        self.data_label: Gtk.Label = gtk_builder.get_object(
+            'thisdevice_data_label')
+
+        hcl_check = subprocess.check_output(['qubes-hcl-report']).decode()
+
+        pattern = re.compile(
+            r"Qubes release\s*(?P<qubes>.+)[\n.]*Brand:\s*(?P<brand>.+)[\n.]*"
+            r"Model:\s*(?P<model>.+)[\n.]*BIOS:\s*(?P<bios>.*)[\n.]+"
+            r"Xen:\s*(?P<xen>.+)[\n.]*Kernel:\s+(?P<kernel>.+)[\n.]*"
+            r"RAM:\s+(?P<ram>.+)[\n.]+CPU:\s*(?P<cpu>.*)[\n.]+"
+            r"Chipset:\s*(?P<chipset>.*)[\n.]+VGA:\s*(?P<vga>.*)")
+        match = pattern.search(hcl_check)
+        if not match:
+            label_text = hcl_check
+            self.data_label.get_style_context().add_class('red_code')
+        else:
+            label_text = f"""<b>Brand:</b> {match.group('brand')}
+<b>Model:</b> {match.group('model')}
+        
+<b>CPU:</b> {match.group('cpu')}
+<b>Chipset:</b> {match.group('chipset')}
+<b>Graphics:</b> {match.group('vga')}
+
+<b>RAM:</b> {match.group('ram')}
+
+<b>QubesOS version:</b> {match.group('qubes')}
+<b>BIOS:</b> {match.group('bios')}
+<b>Kernel:</b> {match.group('kernel')}
+<b>Xen:</b> {match.group('xen')}
+"""
+        self.data_label.set_markup(label_text)
```

Comment:
Add a button to copy this info to the clipboard? Possibly also to call `qubes-hcl-report` and copy its output to the clipboard too.